TypeSystem/type-variance/declaration_A01_t06.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if a variance modifier is specified
+/// for a type parameter declared by a static extension, a generic function
+/// type, a generic function or method, or a type alias.
+///
+/// @description Check that it is a compile-time error if a variance modifier is
+/// specified for a type parameter declared by a type alias
```

Comment:
Could this be a near-duplicate of TypeSystem/type-variance/declaration_A01_t02.dart?

Reply:
It could. But let's have this test (at least we are checking non-function types here unlike `declaration_A01_t02.dart`)?

Reply:
Good point, let's keep it!